src/fty_nut_command_server.cc

```diff
@@ -0,0 +1,326 @@
+        ZstrGuard cmd(zmsg_popstr(msg));
+        ZstrGuard data(zmsg_popstr(msg));
+
+        // Check if command is known in advance (try to prevent failures halfway)
+        if (!valid_commands.count(cmd.get())) {
+            throw std::runtime_error("CMD-NOT-SUPPORTED");
+        }
+        // XXX: we don't handle arguments just yet
+        if (!streq(data, "")) {
+            throw std::runtime_error("CMD-NOT-SUPPORTED");
+        }
+
+        commands.emplace_back(cmd.get(), data.get());
+    }
+
+    const std::string prefix = device.daisy_chain ? std::string("device.") + std::to_string(device.daisy_chain) + "." : "";
+    for (const auto &i : commands) {
+        nut.executeDeviceCommand(device.name, prefix + i.first);
+    }
+
+    zmsg_t *reply = zmsg_new();
+    zmsg_addstr(reply, "OK");
+    zmsg_addstr(reply, uuid);
+    send_reply(client, address, "DO_COMMANDS", &reply);
+}
+
+//
+// Mainloop
+//
+
+void
+fty_nut_command_server(zsock_t *pipe, void *args)
+{
+    std::string dbURL;
+    std::string nutHost = "localhost";
+    std::string nutUsername;
+    std::string nutPassword;
+
+    // Connect to Malamute
+    MlmClientGuard client(mlm_client_new());
+    if (!client) {
+        log_error("mlm_client_new() failed");
+        return;
+    }
+    const char *endpoint = static_cast<const char*>(args);
+    if (mlm_client_connect(client, endpoint, 5000, ACTOR_COMMAND_NAME) < 0) {
+        log_error("client %s failed to connect", ACTOR_COMMAND_NAME);
+        return;
+    }
+
+    // Enter mainloop
+    ZpollerGuard poller(zpoller_new(pipe, mlm_client_msgpipe(client), nullptr));
+    zsock_signal(pipe, 0);
+    while (!zsys_interrupted) {
+        void *which = zpoller_wait(poller, -1);
+        if (zsys_interrupted) {
+            break;
+        }
+        if (which == pipe) {
+            ZmsgGuard msg(zmsg_recv(pipe));
+            ZstrGuard actor_command(zmsg_popstr(msg));
+
+            //  $TERM actor command implementation is required by zactor_t interface
+            if (streq(actor_command, "$TERM")) {
+                return;
+            }
+            else if (streq(actor_command, "NUT_SERVER")) {
+                ZstrGuard host(zmsg_popstr(msg));
+                ZstrGuard username(zmsg_popstr(msg));
+                ZstrGuard password(zmsg_popstr(msg));
+                nutHost = host.get();
+                nutUsername = username.get();
+                nutPassword = password.get();
+                log_info("NUT server '%s' configured", host.get());
+                continue;
+            }
+            else if (streq(actor_command, "DB_URL")) {
+                ZstrGuard databaseURL(zmsg_popstr(msg));
+                dbURL = databaseURL.get();
+                log_info("Database URL configured");
+                continue;
+            }
+            else {
+                log_error("Unrecognized pipe request '%s'", actor_command.get());
+                continue;
+            }
+        }
+
```

Comment:
before doing mlm_client_recv, it's better to ckeck if we got a msg from mlm client.
else if (which == mlm_client_msgpipe (client) {
     ZmsgGuard msg(mlm_client_recv(client));
     ..